### PR TITLE
Migrating Documentation to GitHub page

### PR DESCRIPTION
**Pull request by hayleycd**
We are migrating OSV.dev documentation from redocly to a GitHub page.

- This migration makes future changes to the docs much easier. 
- The Redocly docs have been deleted as part of this migration. 
- Content has been moved from our readme to the docs as well. 
- The FAQ has been moved into the docs. 
- This PR assumes that the docs will be hosted at google.github.io/osv.dev, though we should be able to change this in the future if we want to. 
- google.github.io/osv.dev can be set up immediately after merging. 
- Please check out the redirects to see if they look ok. They should be tried on stage before releasing. 

Additionally, this addresses issues #466 and #573

EDIT: [Link to proposed docs](https://hayleycd.github.io/osv.dev/)

### CODE REVIEW

**Review by oliverchang — COMMENTED** [review]
Thanks @hayleycd! 

Is this documentation site staged somewhere for us to look at?

Also, could we separate the App Engine changes into another PR? We can do this in stages -- first we publish the documentation site, then we move the redirects.

File: docs/faq.md [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+
+1. [The OSV Schema](https://ossf.github.io/osv-schema/): An easy-to-use data format that maps precisely to open source versioning schemes.
+2. Reference infrastructure (this site, API, and tooling) that aggregates and indexes vulnerability data from databases that use the OSV schema.
+
+We created OSV to address many of the shortcomings of dealing with vulnerabilities in open source software using existing solutions.
+
+See our blog posts for more details:
+
+1. [Launching OSV](https://security.googleblog.com/2021/02/launching-osv-better-vulnerability.html)
+2. [Announcing a unified vulnerability schema for open source](https://security.googleblog.com/2021/06/announcing-unified-vulnerability-schema.html)
```
Comment (andrewpollock) [inline docs/faq.md:31]: Let's add our most recent one as well...

```suggestion
2. [Announcing a unified vulnerability schema for open source](https://security.googleblog.com/2021/06/announcing-unified-vulnerability-schema.html)
3. [[OSV and the Vulnerability Life Cycle](https://security.googleblog.com/2023/03/osv-and-vulnerability-life-cycle.html)](https://security.googleblog.com/2023/03/osv-and-vulnerability-life-cycle.html)
```

File: docs/faq.md [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+
+## How do I use OSV as a vulnerability database maintainer?
+By making your vulnerability database available in the OSV format, open source users will have a consistent way to consume vulnerabilities across all open source ecosystems.
+
+Vulnerability databases can also benefit from easier interchange and vulnerabiltiy sharing from other databases that use the OSV format.
+
+## Is the database available to download?
+Yes!
+
+The database in available in a GCS bucket maintained by OSV: [gs://osv-vulnerabilities](https://osv-vulnerabilities.storage.googleapis.com/)
```
Comment (andrewpollock) [inline docs/faq.md:66]: The bucket is _way_ more user friendly for browsing via the Cloud Console, with the caveat that you have to log in...

```suggestion
The database in available in a GCS bucket maintained by OSV: [gs://osv-vulnerabilities](https://osv-vulnerabilities.storage.googleapis.com/) (also [publicly browseable via the Google Cloud Console](https://console.cloud.google.com/storage/browser/osv-vulnerabilities) with a login)
```

File: docs/index.md [hunk 3]
```diff
@@ -0,0 +1,16 @@
+---
+
```
Comment (andrewpollock) [inline docs/index.md:2]: It looks like there might be an errant blank line here?

File: gcp/appengine/frontend3/src/templates/home.html [hunk 4]
```diff
@@ -134,7 +134,7 @@ <h2 class="heading">Use the API</h2>
       <div class="code-card-container mdc-layout-grid__inner">
         <div class="code-card mdc-layout-grid__cell--span-6 mdc-layout-grid__cell--span-12-tablet">
           <h3 class="code-card-title">Query by commit hash</h3>
-          <pre class="code-card-content" id="example-commit-hash">curl -X POST -d \
+          <pre class="code-card-content" id="example-commit-hash">curl POST -d \
```
Comment (andrewpollock) [inline gcp/appengine/frontend3/src/templates/home.html:137]: Looks like there was some de-X-ing here without the next word

```suggestion
          <pre class="code-card-content" id="example-commit-hash">curl -d \
```

File: gcp/appengine/frontend3/src/templates/home.html [hunk 5]
```diff
@@ -143,7 +143,7 @@ <h3 class="code-card-title">Query by commit hash</h3>
         </div>
         <div class="code-card mdc-layout-grid__cell--span-6 mdc-layout-grid__cell--span-12-tablet">
           <h3 class="code-card-title">Query by version number</h3>
-          <pre class="code-card-content" id="example-version-number">curl -X POST -d \
+          <pre class="code-card-content" id="example-version-number">curl POST -d \
```
Comment (andrewpollock) [inline gcp/appengine/frontend3/src/templates/home.html:146]: Same here

```suggestion
          <pre class="code-card-content" id="example-version-number">curl -d \
```

**Review by andrewpollock — APPROVED** [review]
A few minor things/suggestions, otherwise LGTM.

Is there a way to run a link check over everything (either pre-commit, or afterwards) to flush out any broken links inadvertently created?

**Comment by hayleycd** [comment]
@oliverchang 

Yes, of course. It is the same as the proof of concept from earlier. [Docs preview](hayleycd.github.io/osv.dev)

File: docs/index.md (hunk 3 above)
Comment (hayleycd) [inline docs/index.md:2]:
```suggestion
```

**Comment by hayleycd** [comment]
> Thanks @hayleycd!
> 
> Is this documentation site staged somewhere for us to look at?
> 
> Also, could we separate the App Engine changes into another PR? We can do this in stages -- first we publish the documentation site, then we move the redirects.

We can absolutely do this. But there are some unintended consequences that you should know. 

1). Weirdly, you can only select to build a github page from [two potential directories](https://docs.github.com/en/pages/getting-started-with-github-pages/configuring-a-publishing-source-for-your-github-pages-site#about-publishing-sources)--root or `/docs`. 
2). Old documentation was living in `/docs', so either old and new docs live there for a time OR I put all of the new docs in the root which seemed overly chaotic and difficult to evaluate. 
3). At some point in this process, I had old and new docs living together in `/docs` but there were some unintended consequences for the github page (some md files that were part of the "old docs" were trying to render). I am unsure if there would be any unintended consequences to our old docs if this in-between state was included in a release.

Because of all of this, it seemed more straightforward to me to delete the old docs, include the redirects, and publish the new docs all at once. Especially because the site build and the docs build are decoupled. We can merge this, see if the docs look the way we want (they will build on merge), and then test the OSV.dev portion on stage before the next release. 

However, I do understand that this is a lot at once, so I am open to breaking them up. What do you think?

**Comment by hayleycd** [comment]
I split up this pull request as suggested. So I am closing it now, and am opening the seperate pull requests. If you want to merge these all at once, we can always reopen this.

**Pull request closed without merging**